### PR TITLE
Add option to circumvent registry and provide extractor definition directly

### LINKED ISSUE #20: Allow `extract` to be called with a pre-defined `Extractor` definition without registry lookup

### PR DESCRIPTION
**Pull request by ml-evs**
Closes #20.

### CODE REVIEW

File: marda_extractors_api/__init__.py [hunk 1]
```diff
@@ -48,6 +48,8 @@ def extract(
     preferred_mode: SupportedExecutionMethod | str = SupportedExecutionMethod.PYTHON,
     install: bool = True,
     use_venv: bool = True,
+    extractor_definition: dict | None = None,
```
Comment (PeterKraus) [inline marda_extractors_api/__init__.py:51]: When was this syntax introduced? Is the required python version consistent with this syntax?

**Review by PeterKraus — APPROVED** [review]
Quick work.

File: marda_extractors_api/__init__.py (hunk 1 above)
Comment (ml-evs) [reply marda_extractors_api/__init__.py:51]: 3.10 so I think we're on solid ground; this does enforce 3.10 support on the extractors too though -- we could consider checking the version numbers and using the (yet to be developed) docker integration in cases where the Python versions are incompatible

Comment (ml-evs) [reply marda_extractors_api/__init__.py:51]: (https://peps.python.org/pep-0604/)

Comment (ml-evs) [reply marda_extractors_api/__init__.py:51]: The problem is that the venvs made by the code in this repo will have the same version as the env its running in. Not trivial to create venvs of another version without also depending on an external tool (e.g. pyenv). That's a separate problem to this though.

File: tests/test_mpr.py [hunk 2]
```diff
@@ -65,6 +65,38 @@ def test_biologic_extract_from_url(tmp_path, test_mpr_urls, test_mprs):
...
+        output_path = f"mpr-{ind}.nc"
+        data = extract(
+            test_mpr,
+            "biologic-mpr",
+            output_path=output_path,
+            preferred_mode="python",
+            install=(ind == 0),
+            extractor_definition={
+                "id": "yadg",
+                "supported_filetypes": [{"id": "biologic-mpr"}],
```
[the remark this answers is not in the document]
Comment (ml-evs) [reply tests/test_mpr.py:79]: It does get fetched in the normal case, and I think I added this as there is a simple check whether the extractor supports the requested filetype

**Pull request merged**